sql/core/src/main/java/org/apache/spark/sql/sources/v2/reader/MicroBatchReader.java

```diff
@@ -0,0 +1,41 @@
+package org.apache.spark.sql.sources.v2.reader;
+
+import org.apache.spark.sql.sources.v2.reader.Offset;
+import org.apache.spark.sql.execution.streaming.BaseStreamingSource;
+
+import java.util.Optional;
+
+/**
+ * A mix-in interface for {@link DataSourceV2Reader}. Data source readers can implement this
+ * interface to indicate they allow micro-batch streaming reads.
+ */
+public interface MicroBatchReader extends DataSourceV2Reader, BaseStreamingSource {
```

Comment:
I feel there is an implicit semantics that should be documented somehow. For the DataSourceV2Reader (that is, the batch reader), the createReadTasks() will be called only once, whereas, for the Continuous/MicroBatchReader, the createReadTasks() will be called multiple times, and every time after calling `setOffset*`.

Reply:
It's my understanding that createReadTasks() can be called multiple times even on batch readers, when for example restarts happen. I think the only semantic is that createReadTasks() has to respect the values provided by set* methods.